kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -2299,4 +2299,12 @@ object KyuubiConf {
       .version("1.7.0")
       .timeConf
       .createWithDefault(Duration.ofSeconds(60).toMillis)
+
+  val ENGINE_POOL_BALANCE_EXPAND_THRESHOLD: ConfigEntry[Int] =
+    buildConf("kyuubi.engine.expand.threshold")
```

Comment:
we don't need this new configuration and `kyuubi.engine.pool.size.threshold` can be reused I think.

This feature eventually will be more flexible, but let's start w/ a minimal step to make the PR get in master first.